pkg/crypto/cryptoutil/kek_labelers.go

```diff
@@ -0,0 +1,72 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cryptoutil
+
+import (
+	"context"
+	"net"
+	"net/url"
+	"strings"
+
+	"go.thethings.network/lorawan-stack/pkg/types"
+)
+
+// ComponentPrefixKEKLabeler is a ComponentKEKLabeler that joins the component prefix, separator and host.
+type ComponentPrefixKEKLabeler struct {
+	Separator string
+}
+
+func hostFromAddr(addr string) (host string) {
+	if url, err := url.Parse(addr); err == nil && url.Host != "" {
```

Comment:
Isn't this magnitudes simpler and clearer?
(assuming the named return variable is removed)
```go
host := addr
url, err := url.Parse(addr)
if err == nil && url.Host != "" {
	host = url.Host
}

h, _, err := net.SplitHostPort(host)
if err == nil {
	return h
}
return host
```